base/applications/mspaint/history.cpp

```diff
@@ -161,6 +161,8 @@ void ImageModel::Crop(int nWidth, int nHeight, int nOffsetX, int nOffsetY)
 
 void ImageModel::SaveImage(LPTSTR lpFileName)
 {
+    DeleteObject(hBms[(currInd + 1) % HISTORYSIZE]);
+    hBms[(currInd + 1) % HISTORYSIZE] = CreateDIBWithProperties(GetWidth(), GetHeight());
```

Comment:
I'm not quite sure what this does, to be honest. Why does SaveImage modify the history?

Reply:
I'm want be honest, i'm trying to solve this bug without having in-depth knowledge about it. So my solution is like a workaround.
Those 2 lines that you read, just define again the properties of image, because when you use for example, a pen, it will automatically change the pixel format to 32bit. So when an user ends the painting, so when for example you save the image and so phisically save it, it will save with 1bpp color depth. It seems to works, i have tested it.

It changes the history but not update the counter, so when you undo the operation after save, it will undo the last paiting operation.

Reply:
I'm want be honest, i'm trying to solve this bug without having in-depth knowledge about it. So my solution is like a workaround.
Those 2 lines that you read, just define again the properties of image, because when you start drawing, for example using pen, it will automatically change the pixel format to 32bit. So when an user finish to draw and wants to save the image and so phisically save it to disk, with this workaround, it will save with 1bpp color depth. 
It seems to works, i have tested it.

It changes the history but not update the counter, so when you undo the operation after save, it will undo the last paiting operation.